src/saltext/vmware/states/vmc_security_groups.py

```diff
@@ -0,0 +1,421 @@
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        domain_id=domain_id,
+        security_group_id=security_group_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    existing_security_group = None
+
+    if "error" not in get_security_group:
+        log.info("Security group found with Id %s", security_group_id)
+        existing_security_group = get_security_group
+    elif SECURITY_GROUP_NOT_FOUND_ERROR not in get_security_group["error"]:
+        return vmc_state._create_state_response(
+            name=name, comment=get_security_group["error"], result=False
+        )
+
+    if __opts__.get("test"):
+        log.info("present is called with test option")
+        if existing_security_group:
+            return vmc_state._create_state_response(
+                name=name,
+                comment="State present will update Security group {}".format(security_group_id),
+            )
+        else:
+            return vmc_state._create_state_response(
+                name=name,
+                comment="State present will create Security group {}".format(security_group_id),
+            )
+
+    if existing_security_group:
+        updatable_keys = input_dict.keys()
+        is_update_required = vmc_state._check_for_updates(
+            existing_security_group, input_dict, updatable_keys, ["tags"]
+        )
+
+        if is_update_required:
+            updated_security_group = __salt__["vmc_security_groups.update"](
+                hostname=hostname,
+                refresh_key=refresh_key,
+                authorization_host=authorization_host,
+                org_id=org_id,
+                sddc_id=sddc_id,
+                domain_id=domain_id,
+                security_group_id=security_group_id,
+                verify_ssl=verify_ssl,
+                cert=cert,
+                expression=expression,
+                description=description,
+                tags=tags,
+                display_name=display_name,
+            )
+
+            if "error" in updated_security_group:
+                return vmc_state._create_state_response(
+                    name=name, comment=updated_security_group["error"], result=False
+                )
+
+            get_security_group_after_update = __salt__["vmc_security_groups.get_by_id"](
+                hostname=hostname,
+                refresh_key=refresh_key,
+                authorization_host=authorization_host,
+                org_id=org_id,
+                sddc_id=sddc_id,
+                domain_id=domain_id,
+                security_group_id=security_group_id,
+                verify_ssl=verify_ssl,
+                cert=cert,
+            )
+
+            if "error" in get_security_group_after_update:
+                return vmc_state._create_state_response(
+                    name=name, comment=get_security_group_after_update["error"], result=False
+                )
```

Comment:
**suggestion** it still makes sense that `_create_state_response` just gets the response and sets the result itself.

Reply:
this point has already been discussed earlier in other PRs, as it will complicate the things, we agreed on not changing this approach.